Tribler/Core/Modules/MetadataStore/store.py

```diff
@@ -143,98 +149,161 @@ def __init__(self, db_filename, channels_dir, my_key):
         self.clock.init_clock()
 
     def shutdown(self):
+        self._shutting_down = True
         self._db.disconnect()
 
-    def process_channel_dir(self, dirname, channel_id):
+    def process_channel_dir(self, dirname, channel_id, external_thread=True):
         """
         Load all metadata blobs in a given directory.
         :param dirname: The directory containing the metadata blobs.
+        :param external_thread: indicate to lower levels that this is running on a background thread
         :param channel_id: public_key of the channel.
         """
-        # We use multiple separate db_sessions here to limit memory usage when reading big channels
+        # We use multiple separate db_sessions here to limit the memory and reactor time impact,
```

Comment:
This comment applies to the whole method, why do you think it should be moved?

Reply:
I actually meant to respond to the second line of comments (talking about the existence of channels). But on second thought, I guess this is fine since the if statement is only a few lines below the comment.